tests/ttnn/sweep_tests/sweeps/sweeps/transformer_split_query_key_value_and_split_heads.py

```diff
@@ -9,48 +9,313 @@
 import ttnn
 
 from tests.ttnn.utils_for_testing import check_with_pcc
-from models.utility_functions import torch_random
+from models.utility_functions import is_wormhole_b0, is_grayskull
 
+# use combinations of batch_size/core height and q_heads/kv_heads/core width to keep permutations under control
+# some failures are known (e.g. batch_size > cores_h, seq_q > seq_kv, num_kv_heads != num_q_heads when transpose = true) though they shouldn't be failures
+# try to minimize the number of permutations of known failures that shouldn't fail to keep test quick
+# interleaved tests are all expected to fail since the input format is different for sharded and interleaved, and the test mimicks the sharded path
+# they need to be changed to match the sharded path
 
 parameters = {
-    "batch_size": [1],
-    "sequence_size": [384, 1024],
-    "num_heads": [4, 16],
-    "head_size": [64, 128],
-    "input_dtype": [ttnn.bfloat16],
-    "input_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
+    "batch_size_cores_h": [(2, 2), (7, 7), (4, 2)],  # 3  [batch=1] case also needed
+    "seq_len_q_kv": [
+        (64, 64),
+        (256, 96),
+        (64, 96),
+    ],  # 3 [seq_q = seq_kv = 224, 384, and seq_q = 1024, 4096, seq_kv = 96] cases needed by BERT, SD, falcon
+    "num_q_kv_heads_cores_w": [
+        (8, 8, 8),
+        (4, 4, 2),
+        (16, 8, 8),
+    ],  # 3 [q_heads = kv_heads = 12] cases also used in assorted models
+    "head_dim": [64, 160],  # 2 [96, 128] also used
+    "input_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],  # 2
+    "transpose_k": [True],  # 1
+    "separate_tensors": [False, True],  # 2
+    "input_memory_config": [ttnn.L1_BLOCK_SHARDED_MEMORY_CONFIG, ttnn.DRAM_MEMORY_CONFIG],  # 2
 }
 
 
-def run(
-    batch_size, num_heads, sequence_size, head_size, input_dtype, input_memory_config, *, device
+def dtype_size(dtype):
```

Comment:
There should already be a util for this. Otherwise we should create it. 

For bfp, we can also calculate exact size in many cases, which I guess is the overall intent.